script.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * This function creates drag&drop multiple input field,
+ * makes each of "drops" <a> tags in <img> tags,
+ * and then creates a simple gallery out of the loaded images.
+ */
+
+var galleryModule = {
+  makeGallery: function(imageName, imageSrc) {
+    var gallery = document.getElementById("gallery");
+    var image = galleryModule.createImage(imageName, imageSrc); //image = return loadedImage from function createImage()
+    var link = galleryModule.createLink(imageSrc); //link = return loadedLink from function createLink()
+    link.appendChild(image); //append image as the last child of an <a> tag
+    gallery.appendChild(link); //append <a> tag as the last child of the gallery
+  },
+  createImage: function(imageName, imageSrc) {
+    var loadedImage = document.createElement("img"); //create new image object
+    loadedImage.setAttribute("height", imageProcessor.parameters.thumbHeight); //add attributes to the image: size, title, source
+    loadedImage.setAttribute("width", imageProcessor.parameters.thumbWidth);
+    loadedImage.setAttribute("title", imageName);
+    loadedImage.setAttribute("src", imageSrc);
+    return loadedImage;
+  },
+  createLink: function(imageSrc) {
+    var loadedLink = document.createElement("a"); //create <a> tag for hyperlink
+    loadedLink.setAttribute("target", "_blank"); //image will open in new window
+    loadedLink.setAttribute("href", imageSrc); //URL of the page
+    return loadedLink;
+  }
+}
+
+var eventHandlerModule = {
+  attachEvent: function(target, event, callback) {
+    target.addEventListener(event, callback, false);
+  },
+  detachEvent: function(target, event, callback) {
+    target.removeEventListener(event, callback, false);
+  }
+}
+
+var fileModule = {
+  handleFiles: function(querySel) {
+    var files = document.querySelector(querySel).files; //variable, which shows how many files is loading
+    if (files) { 
+      [].forEach.call(files, fileModule.reading); //for each file, which you load is made loadImage function (multiple files)
+    }
+  },
+  reading: function(image) {

```

Comment:
w sposob w jaki napisalas te funkcje sprawia, ze ten komponent jest zupelnie nie reuzywalny i jest na stale przyspawany do gallery module.

to co powinnas zrobic to:
. wyrzucic logike sprawdzania pliku do innej funkcji i uzyc jej jako funkcji pomocniczej (single-responsibility)
. przyjac jako argument allowedFiles a nie brac je ze zmiennej globalnej
. przyjac makeGallery jako callback zamiast hardcodowac to

czyli powinno to wygladac mniej wiecej tak:
```javascript
// nie mowie o image, czy czymkolwiek co jest application specific, 
// wszystkie moje nazwy sa nazwami na tyle ogolnymi, zeby mozna bylo uzywac ich w kilku case'ach
function loadFile(file, allowedFileTypes, callback) {
  // guard pattern z funkcja pomocnicza
  if (!isFileAllowed(file, allowedFileTypes) { return; }

  var reader = new FileReader();
  // ten eventHandlerModule nie jest Ci tu w ogole potrzebny
  reader.onload = function(event) {
    callback(event.target.result);
  };
  reader.readAsDataURL(file);
}
```